imod/prepare/cleanup.py

```diff
@@ -0,0 +1,219 @@
+"""Cleanup utilities"""
+
+from enum import Enum
+from typing import Optional
+
+import xarray as xr
+
+from imod.mf6.utilities.mask import mask_arrays
+from imod.schemata import scalar_None
+from imod.typing import GridDataArray, GridDataset
+
+
+class AlignLevelsMode(Enum):
+    TOPDOWN = 0
+    BOTTOMUP = 1
+
+
+def align_nodata(grids: dict[str, xr.DataArray]) -> dict[str, xr.DataArray]:
+    return mask_arrays(grids)
+
+
+def align_interface_levels(
+    top: GridDataArray,
+    bottom: GridDataArray,
+    method: AlignLevelsMode = AlignLevelsMode.TOPDOWN,
+) -> tuple[GridDataArray, GridDataArray]:
+    to_align = top < bottom
+
+    match method:
+        case AlignLevelsMode.BOTTOMUP:
+            return top.where(~to_align, bottom), bottom
+        case AlignLevelsMode.TOPDOWN:
+            return top, bottom.where(~to_align, top)
+        case _:
+            raise TypeError(f"Unmatched case for method, got {method}")
+
+
+def _cleanup_robin_boundary(
+    idomain: GridDataArray, grids: dict[str, GridDataArray]
+) -> dict[str, GridDataArray]:
+    """Cleanup robin boundary condition (i.e. bc with conductance)"""
+    active = idomain == 1
+    # Deactivate conductance cells outside active domain; this nodata
+    # inconsistency will be aligned in the final call to align_nodata
+    conductance = grids["conductance"].where(active)
+    concentration = grids["concentration"]
+    # Make conductance cells with erronous values inactive
+    grids["conductance"] = conductance.where(conductance > 0.0)
+    # Clip negative concentration cells to 0.0
+    if (concentration is not None) and not scalar_None(concentration):
+        grids["concentration"] = concentration.clip(min=0.0)
```

Comment:
Should we inform the user by producing a warning? Or/and should we log this?

Reply:
For now, I think using the ``standard_log_decorator`` should be sufficient. I added this to the functions. This can be a guidance to the API docs, which will provide more info on what is exactly cleaned up in the function.